clouddrift/ragged.py

```diff
@@ -650,19 +665,26 @@ def subset(
     )
 
     for key in criteria.keys():
-        if key in ds or key in ds.dims:
-            if ds[key].dims == (traj_dim_name,):
+        if np.any(np.isin(key, ds.variables) | np.isin(key, ds.dims)):
+            if isinstance(key, tuple):
+                criterion = [ds[k] for k in key]
+                criterion_dims = criterion[0].dims
```

Comment:
just one think here. I will try to find the time to test later, but since you do `criterion[0].dims`, is it assuming that all elements of the tuple have the same dimension? 

For example, using `datasets.gdp1h` would it work doing with a `Callable` like `def filter(start_lon, ve)` or `def filter(traj, id)`, or `def filter(traj, id, ve)` ?

Reply:
Yes that's the assumption. So indeed your examples would not work. If you agree with the idea, the documentation should probably be clear on this.

Reply:
I can't really think why/how one would use variable with size obs and traj in a single callable.. and we do need to know the length of the result to assign it to the proper dimension.

I think it could be better to validate that all variables of the tuple share the same dimension, and we can return an Exception if it's not case.

Reply:
Yes, it seems the way to go!